callstack/react-native-pager-view
# feat(iOS): Add a `useLegacy` flag to switch between the old/new iOS implementation

Continues #763 

To have a more complete understanding of the issues @troZee ran into while working on #763 and of the scope of the task, I started the implementation from scratch, basing my work on the two (old/new architecture) `UIPageViewController` -> `UIScrollView` re-writes by @krozniata and @okwasniewski directly, and ended up using the previous PR only as a reference. Because of that, to make things more readable, I'm leaving #763 as is and I'm creating a new PR that continues the work started there.

# Summary




- Added the `useLegacy` flag to switch between the used implementation on iOS (old/legacy - `UIPageViewController`, new - `UIScrollView`).
- Adjusted the examples and styles accordingly.
- Reorganized the folder structure in some places to make codegen work and to increase readability.
- Updated the readme file.


## Test Plan

- Run the example on both architectures
- Test both implementations (Legacy vs Basic example) on each architecture
- Check if all functionality is working as expected and the app doesn't crash

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅*     |

(*Just stub code to make it work, the feature doesn't affect Android per se.)

## Checklist
